### PR TITLE
added the tanh_ to paddle tensor frontend

### LINKED ISSUE #22298: tanh_
To-do list: #15115

### PR DESCRIPTION
**Pull request by samthakur587**
# added the tanh_ to paddle tensor frontend

## in this PR i have added the tanh_ to paddle frontend tensor by calling the tanh it self from the same tensor class. it passing all the the backend tests.
image

Close #22298

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 1]
```diff
@@ -3659,6 +3659,40 @@ def test_paddle_tensor_tanh(
...
+def test_paddle_tensor_tanh_(
+    dtype_and_x,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    frontend,
+    on_device,
+    backend_fw,
+):
+  input_dtype, x = dtype_and_x
```
Comment (HaiderSultanArc) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:3680]: There's a linting issue here. Can you please fix this?

Comment (HaiderSultanArc) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:3680]: I also need to see the latest test results. Can you merge main branch in your branch and run the tests again?

Comment (samthakur587) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py]: solved the lint error and tested with latest main branch pull. it is passing all the tests 😄 

image

mergeeee.. 😄 🚀

Comment (HaiderSultanArc) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py]: @vedpatwardhan this is also one of those PRs where CI fails but tests pass locally 🤔 . What could be the issue?

**Review by vedpatwardhan — COMMENTED** [review]
I'd suggest running the tests with a larger number of examples just to confirm that it passes even then. Thanks @HaiderSultanArc @samthakur587 😄
(Given that the tests aren't running in the CI due to a docker issue, we can get the PR merged if you think all the changes are sensible @HaiderSultanArc)

**Comment by samthakur587** [comment]
> I'd suggest running the tests with a larger number of examples just to confirm that it passes even then. Thanks @HaiderSultanArc @samthakur587 😄 (Given that the tests aren't running in the CI due to a docker issue, we can get the PR merged if you think all the changes are sensible @HaiderSultanArc)


Hi @vedpatwardhan @HaiderSultanArc  i always used to test every PR on large number of example in between 100 to 150. i already have tested this function on 150 max examples it passing all the test. i think it should be enough to test this function.

**Comment by samthakur587** [comment]
this is result with 200 examples - 

image

**Comment by samthakur587** [comment]
Hi @vedpatwardhan and @HaiderSultanArc  the error is occuring bcz the with_supported_device_and_dtypes is not define in ivy/backend/paddle/statistic.py file now it's solved.

**Comment by HaiderSultanArc** [comment]
But you said tests were passing locally 🤔 . How did they pass locally if that's the case?

**Comment by samthakur587** [comment]
> But you said tests were passing locally 🤔 . How did they pass locally if that's the case?

yesterday i have pull the latest changes it still passing. but someone made changes today i think and when i pulled the todays changes then this error is comming.

**Comment by samthakur587** [comment]
@vedpatwardhan @HaiderSultanArc  it's still falling 😄  in ```dill``` is not finding i have install the ```dill``` in locally but i thing the problem in docker testing CI. here i think the ```dill``` has not installed.

**Pull request merged**